applications/tari_app_grpc/src/conversions/transaction_output.rs

```diff
@@ -56,15 +56,16 @@ impl TryFrom<grpc::TransactionOutput> for TransactionOutput {
             .try_into()
             .map_err(|_| "Metadata signature could not be converted".to_string())?;
         let covenant = Covenant::consensus_decode(&mut output.covenant.as_slice()).map_err(|err| err.to_string())?;
-        Ok(Self {
+        Ok(Self::new(
+            TransactionOutputVersion::try_from(output.version as u8)?,
```

Comment:
Should check for u8 overflows
```suggestion
            TransactionOutputVersion::try_from(u8::try_from(output.version).map_err(|_| "Invalid version: overflowed u8"))?,
```